File: app/src/main/java/com/kee0kai/thekey/room/dao/StorageFilesDao.java
package com.kee0kai.thekey.room.dao;

import static androidx.room.OnConflictStrategy.REPLACE;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.kee0kai.thekey.room.model.StorageFileEntry;

@Dao
public interface StorageFilesDao {

    String TABLE_NAME = "storagefiles";

    @Insert(onConflict = REPLACE)
    void insert(StorageFileEntry entry);

    @Query("SELECT COUNT(*)>0 FROM " + StorageFilesDao.TABLE_NAME + " WHERE path = :file")
    boolean exist(String file);

    @Query("SELECT COUNT(*) FROM " + StorageFilesDao.TABLE_NAME)
    int count();

    @Query("DELETE  FROM " + StorageFilesDao.TABLE_NAME + " WHERE path = :file")
    void delete(String file);

    @Query("DELETE FROM " + StorageFilesDao.TABLE_NAME)
    void deleteAll();

    @Query("SELECT * FROM " + StorageFilesDao.TABLE_NAME)
    StorageFileEntry[] get();

    @Query("SELECT * FROM " + StorageFilesDao.TABLE_NAME + " WHERE path=:path")
    StorageFileEntry get(String path);

}

File: app_mobile/src/main/kotlin/com/github/klee0kai/thekey/app/data/room/dao/StorageFilesDao.kt
package com.github.klee0kai.thekey.app.data.room.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.github.klee0kai.thekey.app.data.room.entry.StorageFileEntry

@Dao
interface StorageFilesDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertOrUpdate(entry: StorageFileEntry)

    @Query("SELECT COUNT(*)>0 FROM $TABLE_NAME WHERE path = :file")
    fun exist(file: String?): Boolean

    @Query("SELECT COUNT(*) FROM $TABLE_NAME")
    fun count(): Int

    @Query("DELETE  FROM $TABLE_NAME WHERE path = :file")
    fun delete(file: String?)

    @Query("DELETE FROM $TABLE_NAME")
    fun deleteAll()

    @Query("SELECT * FROM $TABLE_NAME")
    fun getAll(): List<StorageFileEntry>

    @Query("SELECT * FROM $TABLE_NAME WHERE path=:path")
    fun get(path: String?): StorageFileEntry?

    companion object {
        const val TABLE_NAME = "storagefiles"
    }
}